create_images.py

```diff
@@ -169,36 +175,50 @@ def add_vline(self, axis, index):
 
 def main(args):
     files = args.filename
-    output_filename = os.path.realpath(args.output)
     logger.info('Building {} files'.format(len(files)))
+    if args.output is None:
+        logger.warning('Not creating movie')
 
-    with temporary_directory() as image_dir:
+    delete = not args.no_delete_tmpdir
+    with temporary_directory(args.images_dir, delete=delete) as image_dir:
         logger.info("Building into {}".format(image_dir))
 
-        sorted_files = files if args.no_sort else sort_images(files)
+        if args.no_sort:
+            logger.warning('Not sorting images')
+            sorted_files = files
+        else:
+            sorted_files = sort_images(files)
+
         logger.info('Extracting time series from data')
         median_behaviour = TimeSeries.extract_from(sorted_files)
 
         logger.info('Building movie images')
         pool = mp.Pool() if not args.no_multiprocessing else NullPool()
-        pool.map(partial(build_image, outdir=image_dir,
-                         median_behaviour=median_behaviour),
-                 enumerate(sorted_files))
+        fn = partial(build_image, outdir=image_dir,
+                     median_behaviour=median_behaviour)
+        pool.map(fn, enumerate(sorted_files))
 
-        generate_movie(image_dir, output_filename, fps=args.fps)
+        if args.output is not None:
+            output_filename = os.path.realpath(args.output)
+            generate_movie(image_dir, output_filename, fps=args.fps)
 
 
 def parse_args():
     parser = argparse.ArgumentParser()
     parser.add_argument('filename', nargs='+')
     parser.add_argument('-o', '--output', help="Output movie name",
-                        required=False, default='output.mp4')
+                        required=False)
```

Comment:
I'm not sure if this is the best way to approach this. It may lead to surprising behaviour if the user forgets to supply the output argument.